### ISSUE TITLE
`E0133` (Unsafe code outside unsafe block)  is not detected when accessing to union field

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#10245]
## Environment

* **IntelliJ Rust plugin version:** 0.4.191.5264-231-nightly
* **Rust toolchain version:** 1.68.0 (2c8cc3432 2023-03-06) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2023.1 Beta Ultimate Edition (IU-231.7864.76)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
https://doc.rust-lang.org/error_codes/E0133.html
<img width="910" alt="image" src="https://user-images.githubusercontent.com/49211026/224976919-00178089-5eb2-4e79-97aa-37ec346e81dc.png">
 
## Steps to reproduce
```rust
union Foo {
    x: u32,
    y: u64
}

fn main() {
    let foo = Foo { x: 22 };
    let  a = foo.x;
}


```
